html-aam/scripts/html-aam-accname-utils.js

```diff
@@ -0,0 +1,338 @@
+      elReturn = elRuby;
+    }
+
+    else if (specialContext === 'rt') {
+      let elRuby = document.createElement('ruby');
+      elRuby.appendChild(document.createTextNode('ruby text node contents'));
+      elRuby.appendChild(document.createElement('rp')).appendChild(document.createTextNode('rp #1'));
+      elRuby.appendChild(elTest);
+      elTest.appendChild(document.createTextNode('rt'));
+      elRuby.appendChild(document.createElement('rp')).appendChild(document.createTextNode('rp #2'));
+      elReturn = elRuby;
+    }
+
+    return elReturn;
+  },
+
+  getUniqueTestId: function () {
+    return this.testId += 1;
+  },
+
+  parseSelector: function (selector) {
+    const regex = /^(\w+)(?:\[(\w+)="([\w-]+)"\])?$/;
+    const [_, element, attrName = null, attrValue = null] = selector.match(regex);
+    return { element, attrName, attrValue };
+  },
+
+  buildPageHeading: function (testConfig) {
+    const elBody = document.querySelector('body');
+
+    if (!document.querySelector('h1')) {
+      const elH1 = document.createElement('h1');
+      elH1.textContent = document.title;
+      elBody.appendChild(elH1);
+    }
+
+    if (testConfig.urlSpec) {
+      elBody.insertAdjacentHTML('beforeend', `<p><a href="${testConfig.urlSpec}">HTML-AAM computation steps</a></p>`);
+    }
+  },
+
+  previewTestMarkup: function (html) {
+    const elBody = document.querySelector('body');
+    let elTestMarkup = document.createElement('pre');
+    elTestMarkup.style.border = '1px solid green';
+    elTestMarkup.style.backgroundColor = 'palegreen';
+    elTestMarkup.style.padding = '1rem';
+    elTestMarkup.style.whiteSpace = 'pre-wrap';
+    elTestMarkup.appendChild(document.createTextNode(html.replaceAll('><', '>\n<').trim()));
+    elBody.appendChild(elTestMarkup);
+  },
+
+
+  testId: 0,
+  srcImgSample: "data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMjgiIGhlaWdodD0iMzYiIGZpbGw9Im5vbmUiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyI+CiAgPHBhdGggZD0iTTIwIDZoLTJ2NGgtMlY2aC00djJoLTJ2MmgydjJIOFY4SDZ2NGgydjRoMTR2LTJoLTZ2LTJoMnYtMmgydjJoNnYtMmgtMlY4aC00VjZaIiBmaWxsPSIjRUZDMjk4Ii8+CiAgPHBhdGggZD0iTTYgNmg2djJoLTJ2MmgydjJIOFY4SDZWNloiIGZpbGw9IiMwMDAiLz4KICA8cGF0aCBkPSJNNiA4SDR2Nmg0di0ySDZWOFptMTAtMmgydjRoLTJWNlptMiA0aDJ2Mmg0djJoLTh2LTJoMnYtMloiIGZpbGw9IiMwMDAiLz4KICA8cGF0aCBkPSJNMiAyMmg0djJoMnYySDZ2Mkgydi02Wm0yMCAwdjJoLTJ2MmgydjJoNHYtNmgtNFoiIGZpbGw9IiNFMUQ3RDYiLz4KICA8cGF0aCBkPSJNNCAzMGg2djRIMnYtMmgydi0yWm0yMCAwaC02djRoOHYtMmgtMnYtMloiIGZpbGw9IiM4RDU3M0QiLz4KICA8cGF0aCBkPSJNMTIgMjJoLTJ2Mmgydi0yWm02IDBoLTJ2Mmgydi0yWiIgZmlsbD0iI0ZDRjEzOCIvPgogIDxwYXRoIGZpbGwtcnVsZT0iZXZlbm9kZCIgY2xpcC1ydWxlPSJldmVub2RkIiBkPSJNMTIgMTZoLTJ2Nkg4djRINnY0aDZ2LTJoNHYyaDZ2LTRoLTJ2LTRoLTJ2LTRoLTJ2MmgtNHYtNFptNiA2djJoLTJ2LTJoMlptLTggMGgydjJoLTJ2LTJaIiBmaWxsPSIjNTY3MkQ2Ii8+CiAgPHBhdGggZD0iTTggMnYySDZ2MmgxOFY0aC00VjJIOFptMTAgMTRoLTZ2NGg0di0yaDJ2NGgydjJoMnYtMmg0di0yaC0ydi0yaC02di0yWk02IDE4SDR2MkgydjJoNHYyaDJ2LTJoMnYtNkg2djJaIiBmaWxsPSIjRDAxQTE5Ii8+Cjwvc3ZnPgo=",
```

Comment:
This "Mario Bros" graphic probably has some copyright restrictions by Nintendo. It's also an SVG, which triggers different rendering pipelines than standard raster images.

FYI... there's a shorter, single-pixel transparent GIF in other tests:
`data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACH5BAEAAAAALAAAAAABAAEAAAICRAEAOw==`

Reply:
So much for my easter egg, `<audio src="sad-trombone-sound.mp3">`.

I should have known better. I’ll use that GIF.

Reply:
[Done](https://github.com/web-platform-tests/wpt/pull/44965/commits/9260cfb2bd48a23e973bbadb71a08391e0fbee74).